runtime/src/accounts_db.rs

```diff
@@ -3249,6 +3249,172 @@ impl AccountsDb {
         total_accounts_after_shrink
     }
 
+    /// Remove all accounts not in `minimized_account_set`. This should only be used by ledger-tool to create minimized snapshots.
+    pub(crate) fn minimize_accounts_db(
```

Comment:
I think this is the only pub function in this group of functions. I wonder if these functions can be in their own file and be given &AccountsDB (if everything they need is at least pub(crate)) or if we should put these functions in their own namespace/impl in this file. I keep trying to make this file not quite so big. And this is very specialized code used only by a very specific option on ledger-tool.

Reply:
Was able to move much of the minimization functionality out of bank and accounts_db, but not all. Some used private functions that shouldn't be directly exposed.

I think another option is to use pub(in snapshot_minimizer) to give access to those functions ONLY in the new snapshot_minimizer module.  I haven't seen that rust feature used elsewhere in the code, so I'm hesitant to take that approach. What do you think?